tests/test_utils.py

```diff
@@ -0,0 +1,143 @@
+from collections import namedtuple
+from contextlib import contextmanager
+from itertools import product
+from tempfile import TemporaryFile
+
+from pytest import fixture
+from yaml import safe_load
+
+from utils.deploy import main as deploy
+
+RESOURCE_TEMPLATES_INDEXES = {
+    "insights-inventory-reaper": 0,
+    "insights-host-delete": 1,
+    "insights-inventory-mq-service": 2,
+    "insights-inventory": 3,
+}
+TARGETS_INDEXES = {"prod": 0, "stage": 1}
+DEPLOY_YML = """$schema: /some-schema
+
+labels:
+  some: labels
+
+resourceTemplates:
+- name: insights-inventory-reaper
+  targets:
+  - namespace:
+      $ref: /host-inventory-prod.yml
+    ref: master
+    parameters:
+      IMAGE_TAG: 626ae85
+      REAPER_SUSPEND: 'true'
+  - namespace:
+      $ref: /stage-host-inventory-stage.yml
+    ref: master
+    parameters:
+      IMAGE_TAG: 626ae85
+      REAPER_SUSPEND: 'true'
+- name: insights-host-delete
+  parameters:
+    REPLICAS: 1
+  targets:
+  - namespace:
+      $ref: /host-inventory-prod.yml
+    ref: master
+    parameters:
+      IMAGE_TAG: 587a4f3
+  - namespace:
+      $ref: /stage-host-inventory-stage.yml
+    ref: master
+    parameters:
+      IMAGE_TAG: 587a4f3
+- name: insights-inventory-mq-service
+  parameters:
+    REPLICAS: 1
+  targets:
+  - namespace:
+      $ref: /host-inventory-prod.yml
+    ref: master
+    parameters:
+      IMAGE_TAG: 626ae85
+  - namespace:
+      $ref: /stage-host-inventory-stage.yml
+    ref: master
+    parameters:
+      IMAGE_TAG: 626ae85
+- name: insights-inventory
+  parameters:
+    REPLICAS: 1
+  targets:
+  - namespace:
+      $ref: /host-inventory-prod.yml
+    ref: master
+    parameters:
+      IMAGE_TAG: 626ae85
+      BULK_QUERY_SOURCE: xjoin
+  - namespace:
+      $ref: /stage-host-inventory-stage.yml
+    ref: master
+    parameters:
+      IMAGE_TAG: 626ae85
+      BULK_QUERY_SOURCE: xjoin
+"""
+
+
+Args = namedtuple("Args", ("promo_code",))
+
+
+@fixture(scope="function")
+def temporary_file():
+    @contextmanager
+    def _fixture():
+        with TemporaryFile("r+", encoding="utf-8") as file:
+            yield file
+
+    return _fixture
+
+
+@fixture(scope="function")
+def run_deploy(temporary_file):
+    def _fixture(promo_code, inp_):
+        args = Args(promo_code)
+
+        with temporary_file() as inp, temporary_file() as outp:
+            inp.write(inp_)
+            inp.seek(0)
+
+            deploy(args, inp, outp)
+
+            outp.seek(0)
+            return outp.read()
+
+    return _fixture
+
+
+@fixture(scope="session")
+def head():
```

Comment:
I also think that we don't need a fixture for this. I could be a simple helper function.
```suggestion
def head(doc):
    lines = doc.split("\n")
    first_lines = lines[0:7]
    return "\n".join(first_lines)
```